### PR TITLE
docs: Helm plugins via initContainers

### PR DESCRIPTION
**Pull request by djfinnoy**
Related: #7066

Include an alternative method for installing Helm plugins that don't require users to maintain their own version of the ArgoCD container image.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/user-guide/helm.md [hunk 1]
```diff
@@ -199,6 +201,60 @@ You have to remember about `HELM_PLUGINS` environment property - this is require
...
+        - name: HELM_CONFIG_HOME
+          value: /helm/config
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /gcloud/key.json
+      command: ["/bin/sh", "-c"]
+      args:
+        - apk --no-cache add curl;
+          helm plugin install https://github.com/hayorov/helm-gcs.git;
+          helm repo add bulder-shared gs://my-private-helm-gcs-repository;
+          chmod -R 777 $HELM_DATA_HOME;
```
Comment (jannfis) [inline docs/user-guide/helm.md:256]: I think this block needs to be wrapped into triple-backticks as a code block in order to display correctly.

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks for improving the docs, @djfinnoy!

Please have a look at my comment below, so that the docs will display correctly.

File: docs/user-guide/helm.md (hunk 1 above)
Comment (djfinnoy) [reply docs/user-guide/helm.md:257]: whoops, my bad

**Review by jannfis — APPROVED** [review]
LGTM, thanks @djfinnoy !

**Pull request merged**